### PR TITLE
feat(wallet): add registration functionality

### LINKED ISSUE #134: Implement registration
- related to #48

### PR DESCRIPTION
**Pull request by raducristianpopa**
### Context
- fixes #134 

### Changes

### CODE REVIEW

File: packages/wallet/src/pages/auth/signup.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      openDialog(
+        <ErrorDialog
+          onClose={closeDialog}
+          content="Something went wrong. Please try again"
+        />
+      )
+      return
+    }
+
+    if (response.success === true) {
```
Comment (Tymmmy) [inline packages/wallet/src/pages/auth/signup.tsx:33]:
```suggestion
    if (response.success) {
```

**Review by Tymmmy — APPROVED** [review]

**Review by Tymmmy — APPROVED** [review]

File: packages/backend/src/index.ts [hunk 2]
```diff
@@ -17,13 +17,23 @@ const app: Application = express()
...
 app.use(express.urlencoded({ extended: true }))
 app.use(cookieParser())
 
 const knex = Knex(config[env.NODE_ENV || 'development'])
 Model.knex(knex)
+;(async () => {
+  knex.migrate.latest({
+    directory: __dirname + '/../migrations'
+  })
+})()
```
Comment (raducristianpopa) [inline packages/backend/src/index.ts:36]: Prettier is formatting like this.

Comment (Tymmmy) [reply packages/backend/src/index.ts:36]: Formatting with the ';' ? Or that can be deleted?

Comment (raducristianpopa) [reply packages/backend/src/index.ts:36]: Yep, it's adding `;`.

File: packages/wallet/src/lib/api/user.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import type { AxiosError } from 'axios'
+import { z } from 'zod'
+import $axios, { type ErrorResponse, type SuccessResponse } from '../axios'
+
+export const signUpSchema = z
+  .object({
+    email: z.string().email({ message: 'Email is required' }),
+    password: z
+      .string()
+      .min(6, { message: 'Password should be at least 6 characters long.' }),
```
Comment (Tymmmy) [inline packages/wallet/src/lib/api/user.ts:10]: SHould we use '.' everywhere? Or not use at all?

**Review by Tymmmy — APPROVED** [review]

**Review by Tymmmy — APPROVED** [review]

**Pull request merged**